internal/controllers/terraformlayer/controller.go

```diff
@@ -70,11 +70,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	locked, err := lock.IsLocked(ctx, r.Client, layer)
 	if err != nil {
 		log.Errorf("failed to get Lease Resource: %s", err)
-		return ctrl.Result{RequeueAfter: r.Config.Controller.Timers.OnError}, err
+		return ctrl.Result{RequeueAfter: layer.GetTimeBeforeRetry(r.Config.Controller.Timers.OnError)}, err
 	}
 	if locked {
 		log.Infof("terraform layer %s is locked, skipping reconciliation.", layer.Name)
-		return ctrl.Result{RequeueAfter: r.Config.Controller.Timers.WaitAction}, nil
+		return ctrl.Result{RequeueAfter: layer.GetTimeBeforeRetry(r.Config.Controller.Timers.WaitAction)}, nil
```

Comment:
Same here, this is not linked to any failure this is only a wait action, we might in the future use a metric for the wait action time depending on the execution time of a layer but for now we don't have this metric